File: automaguard-dotnet/tests/AutomaGuard.Tests/Fixtures.cs
namespace AutomaGuard.Tests;

/// <summary>Shared test fixtures.</summary>
internal static class Fixtures
{
    internal static readonly PolicyResult AllowResult = new()
    {
        Verdict = Verdict.Allow,
        Reason = null,
        TriggeredRules = [],
        Violations = [],
        ConstraintViolations = [],
        Actions = [],
        LatencyMicroseconds = 3,
    };

    internal static readonly PolicyResult DenyResult = new()
    {
        Verdict = Verdict.Deny,
        Reason = "Tool \"exec\" is not allowed",
        TriggeredRules = [0],
        Violations = [],
        ConstraintViolations = [],
        Actions = [],
        LatencyMicroseconds = 4,
    };

    internal static readonly PolicyResult AuditResult = new()
    {
        Verdict = Verdict.Audit,
        Reason = "Flagged for audit",
        TriggeredRules = [1],
        Violations = [],
        ConstraintViolations = [],
        Actions = [],
        LatencyMicroseconds = 2,
    };

    internal static readonly PolicyResult RedactResult = new()
    {
        Verdict = Verdict.Redact,
        Reason = "PII detected in output",
        TriggeredRules = [2],
        Violations = [],
        ConstraintViolations = [],
        Actions = [],
        LatencyMicroseconds = 5,
    };

    internal static readonly EngineStatus DefaultStatus = new()
    {
        PolicyName = "TestPolicy",
        Severity = "High",
        TotalRules = 2,
        TotalStateMachines = 1,
        ActiveStateMachines = 1,
        ViolatedStateMachines = 0,
        SatisfiedStateMachines = 0,
        TotalConstraints = 0,
        EventsProcessed = 0,
    };
}

File: automaguard-java/src/test/java/io/automaguard/Fixtures.java
package io.automaguard;

import java.util.List;

/** Shared test fixtures — pre-built {@link PolicyResult} instances. */
final class Fixtures {

    static final PolicyResult ALLOW_RESULT = new PolicyResult(
            Verdict.ALLOW, null, List.of(), List.of(), List.of(), 3L);

    static final PolicyResult DENY_RESULT = new PolicyResult(
            Verdict.DENY, "Tool \"exec\" is not allowed",
            List.of(0), List.of(), List.of(), 4L);

    static final PolicyResult AUDIT_RESULT = new PolicyResult(
            Verdict.AUDIT, "Flagged for audit",
            List.of(1), List.of(), List.of(), 2L);

    static final PolicyResult REDACT_RESULT = new PolicyResult(
            Verdict.REDACT, "PII detected in output",
            List.of(2), List.of(), List.of(), 5L);

    static final PolicyResult VIOLATION_RESULT = new PolicyResult(
            Verdict.DENY, "NoPIIExfiltration violated",
            List.of(0),
            List.of(new Violation("ExfiltrationGuard", "NoPIIExfiltration",
                    "PII data was followed by an external request")),
            List.of(), 6L);

    static final PolicyResult RATE_LIMIT_RESULT = new PolicyResult(
            Verdict.DENY, "Rate limit exceeded",
            List.of(),
            List.of(),
            List.of(new ConstraintViolation("RateLimit", "data_access", 20L, 21L, 60000L)),
            7L);

    private Fixtures() {}
}
